WooCommerce/Classes/ViewRelated/Orders/OrderListViewModel.swift

```diff
@@ -149,6 +132,40 @@ final class OrderListViewModel {
                                  reason: reason,
                                  completionHandler: completionHandler)
     }
+
+    private func createQuery() -> FetchResultSnapshotsProvider<StorageOrder>.Query {
+        let predicate: NSPredicate = {
+            let excludeSearchCache = NSPredicate(format: "exclusiveForSearch = false")
+            let excludeNonMatchingStatus = statusFilter.map { NSPredicate(format: "statusKey = %@", $0.slug) }
+
+            var predicates = [ excludeSearchCache, excludeNonMatchingStatus ].compactMap { $0 }
+            if !includesFutureOrders, let nextMidnight = Date().nextMidnight() {
+                // Exclude orders on and after midnight of today's date
+                let dateSubPredicate = NSPredicate(format: "dateCreated < %@", nextMidnight as NSDate)
+                predicates.append(dateSubPredicate)
+            }
+
+            return NSCompoundPredicate(andPredicateWithSubpredicates: predicates)
+        }()
+
+        guard let siteID = ServiceLocator.stores.sessionManager.defaultStoreID else {
```

Comment:
nit: Should we inject the `siteID`?

Reply:
yes in a future PR! now that #2850 was merged, there is only one site ID for its lifetime. DI requires more levels of changes, so I plan to save it for a separate PR soon.